Automattic/simplenote-ios
# Issue/1043 add multi select

### Fix
This PR fixes #1043 

This PR is intended to add multiple selection support to Simplenote iOS.  Currently there is not way to select and trash multiple notes at a time in SNiOS.  In this PR the ability to enter an edit mode, select multiple notes at a time and then trash them is now possible!

<img src ="https://cdn-std.droplr.net/files/acc_593859/AszY2Y" />

### Test
1. enable and disable edit mode.  Make sure that the mode animates in and out.  While active ensure that you can select multiple notes, that you can't transition to seeing a note and that entering search mode dismisses edit mode
2. Make sure when you go in and out of edit mode the button in the new note bar at the bottom transitions from the new notice icon to the trash icon
3. While in edit mode, select multiple notes to make sure that the view title changes from the standard note list name to "X Selected" Exit edit mode to ensure that the note list returns to its regular name.
4. Test selecting multiple notes and then using the trash button to trash the notes
5. Test changing the note list tag while in edit mode to ensure that edit mode exits
6. While in edit mode you should see a Select All/Deselect all button in the top left where the main menu usually is.  Enable/disable edit mode to make sure that appears and disappears
7. Tap on the Select All button and confirm all notes are selected.  Tap Deselect and make sure they are deselected
8. Delete a selection of notes to see a new in app notification.  Make sure the correct number of notes is marked as being deleted in the notification.  Test the undo button to make sure that all the bulk deleted notes are restored

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
***(Required)*** Add a concise statement to `RELEASE-NOTES.txt` if the changes should be included in release notes. Include details about updating the notes in this section. For example:
> `RELEASE-NOTES.txt` was updated in d3adb3ef with:
> 
> > Added markdown support

If the changes should not be included in release notes, add a statement to this section. For example:

> These changes do not require release notes.
